### PR TITLE
POST /activities

### PR DESCRIPTION
**Pull request by tschuy**
Fixes https://github.com/osuosl/timesync-node/issues/32

This appears to have been forgotten from an earlier sprint.

### CODE REVIEW

**Comment by LyonesGamer** [comment]
Looks comprehensive and accurate. +1

File: tests/activities.js [hunk 1]
```diff
@@ -234,4 +240,204 @@ module.exports = function(expect, request, baseUrl) {
...
+        expect(body.text).to.equal('Field name of' +
+        ' activity should be string but was sent as array');
+
+        checkListEndpoint(done);
+      });
+    });
+
+    it('fails to create an activity with bad slug datatype', function(done) {
+      requestOptions.form = copyJsonObject(postArg);
+      requestOptions.form.object.name = ['test'];
```
Comment (athai) [inline tests/activities.js]: requestOptions.form.object.slug

**Comment by tschuy** [comment]
Updated. @Kennric @LyonesGamer

File: tests/activities.js [hunk 2]
```diff
@@ -1,5 +1,11 @@
 'use strict';
 
+function copyJsonObject(obj) {
+  // This allows us to change object properties
+  // without effecting other tests
```
Comment (LyonesGamer) [inline tests/activities.js]: Affecting. (Although it is true that it also won't effect the other tests. At least I *really* hope so.)

Comment (tschuy) [inline tests/activities.js]: I blame that on copying the function from elsewhere.

**Comment by Kennric** [comment]
I think this is complete. +1

**Comment by LyonesGamer** [comment]
Looks good to me. +1

File: tests/activities.js [hunk 3]
```diff
@@ -516,4 +516,204 @@ module.exports = function(expect, request, baseUrl) {
...
+    it('fails to create a new activity with an invalid slug', function(done) {
+      const postInvalidSlug = copyJsonObject(postArg);
+      postInvalidSlug.object.slug = '$*#*cat';
+      requestOptions.form = postInvalidSlug;
+
+      request.post(requestOptions, function(err, res, body) {
+        const expectedError = {
+          status: 400,
+          error: 'Bad object',
+          text: 'Field slug of activity should be slug but received ' +
```
Comment (athai) [inline tests/activities.js]: Error messages were changed - https://github.com/osuosl/timesync-node/blob/develop/src/errors.js#L102

**Comment by tschuy** [comment]
@LyonesGamer @Kennric

**Comment by LyonesGamer** [comment]
Looks good. +1

**Comment by Kennric** [comment]
This looks good to me. +1

**Pull request merged**